lib/Twig/Test/IntegrationTestCase.php

```diff
@@ -17,27 +17,52 @@
  */
 abstract class Twig_Test_IntegrationTestCase extends PHPUnit_Framework_TestCase
 {
-    abstract protected function getExtensions();
+    /**
+     * @return string
+     */
     abstract protected function getFixturesDir();
 
     /**
-     * @dataProvider getTests
+     * @return Twig_ExtensionInterface[]
      */
-    public function testIntegration($file, $message, $condition, $templates, $exception, $outputs)
+    protected function getExtensions()
     {
-        $this->doIntegrationTest($file, $message, $condition, $templates, $exception, $outputs);
+        return array();
+    }
+
+    /**
+     * @return Twig_SimpleFilter[]
+     */
+    protected function getFilters()
+    {
+        return array();
     }
 
     /**
-     * @dataProvider getLegacyTests
-     * @group legacy
+     * @return Twig_SimpleFunction[]
      */
-    public function testLegacyIntegration($file, $message, $condition, $templates, $exception, $outputs)
```

Comment:
I see, I didn't know that. 
Could the legacy code testing be moved to its own class, this would lead to a cleaner `IntegrationTestCase` (and history of that file)? Or would that take away the whole point of the test?